src/pint/models/timing_model.py

```diff
@@ -1851,7 +1853,6 @@ def compare(
                         except (ValueError, AttributeError):
                             newstr += " {:28f}".format(otherpar.value)
                         if otherpar.value != par.value:
-                            sys.stdout.flush()
```

Comment:
These flushes were needed because the warnings came out in the wrong order compared to the strings being written to stdout.  Does the new logging mechanism fix this somehow without all the flushes?

Reply:
Logging material may well not be emitted to stdout at all, for example being written to a log file. If you want something on stdout, write it there.

Reply:
That may be a better way to do it, but deleting the flushes without changing the logging to prints is definitely going to produce wrong results.  So I'd say either change the logging to print statements (with color to make them visible among the other output) or restore the flushes and file an issue that there is a better way to do it.

Reply:
Restoring the flushes is a problem because `log.handlers[0].flush` is liable to break if anyone uses non-trivial logging arrangements - the `.handlers` attribute is no part of the documented interface, and goodness knows which one will be element zero in the list. Intermixing logging and writing to stdout and expecting them to appear in the same place in any particular order is a mistake. What was the goal with this intermixing?

Reply:
As a concrete example, if this code were tested, it is unlikely the test would be able to detect the messages in the intended order, because `pytest` redirects both stdout and log messages. (This modified version of the code does not fail any test.)

Reply:
Hmm. It looks like things have changed so maybe this is moot. I'm pretty sure that compare() used to print the output, but now it returns a list of strings, and the user has to print them manually.  So, the flushes don't actually do anything. The way it is now, the warnings all come out first. They never come out close to the line that they refer to, which is bad.  The solution, as you say, is to not use any logging in this code but just add the warning messages to the string. I don't know how to make them stand out with colors in that case, though.

Reply:
Why does the order matter? Do the warnings not contain enough information for us to know what they refer to? That's a problem if they are going into the logs (which the rest of the output isn't).

I don't think we can rely on colours, because (for example) the `TimingModel.compare()` output goes into the summary PDF in a `verbatim` environment.

Reply:
Perhaps this should go into an issue about how `TimingModel.compare` works? It doesn't seem like this PR affects the problem but it sounds like the comparison is not operating ans designed.

Reply:
Yeah, it is fine to delete the flushes since they aren't working.  We can consider separately how compare should work.